pkg/daemon/ceph/client/subvolumegroup.go

```diff
@@ -100,32 +100,31 @@ func PinCephFSSubVolumeGroup(context *clusterd.Context, clusterInfo *ClusterInfo
 
 func validatePinningValues(pinning cephv1.CephFilesystemSubVolumeGroupSpecPinning) error {
 	numNils := 0
-	var err error
+
 	if pinning.Export != nil {
 		numNils++
-		if *pinning.Export > 256 {
-			err = errors.Errorf("validate pinning type failed, Export: value too big %d", *pinning.Export)
-		} else if *pinning.Export < -1 {
-			err = errors.Errorf("validate pinning type failed, Export: negative value %d not allowed except -1", *pinning.Export)
+		if *pinning.Export > 256 || *pinning.Export < -1 {
+			return fmt.Errorf("validate pinning type failed, Export: value must be between -1 and 256, got %d", *pinning.Export)
 		}
 	}
+
 	if pinning.Distributed != nil {
 		numNils++
-		if !(*pinning.Distributed == 1) && !(*pinning.Distributed == 0) {
-			err = errors.Errorf("validate pinning type failed, Distributed: unknown value %d", *pinning.Distributed)
+		if *pinning.Distributed != 0 && *pinning.Distributed != 1 {
+			return fmt.Errorf("validate pinning type failed, Distributed: value must be 0 or 1, got %d", *pinning.Distributed)
```

Comment:
I think no need for the change unless it improves anything,

And with the updated change,  
we look for this only if the config is correct, but this the first priority to be set.
```
	if numNils > 1 {
		return fmt.Errorf("only one can be set")
	}
```

Reply:
Okay, I see so earlier returning is disabled by choice as this error should have priority makes sense, thanks for explanation @parth-gr! Okay, then I will just adjust the error message that has the 1.1 instead of 1.0